x-pack/plugins/siem/public/components/markdown/index.tsx

```diff
@@ -14,33 +14,35 @@ const TableHeader = styled.thead`
   font-weight: bold;
 `;
 
-const markdownRenderers = {
-  root: ({ children }: { children: React.ReactNode[] }) => (
-    <EuiText data-test-subj="markdown-root" grow={false}>
-      {children}
-    </EuiText>
-  ),
-  table: ({ children }: { children: React.ReactNode[] }) => (
-    <table data-test-subj="markdown-table" className="euiTable euiTable--responsive">
-      {children}
-    </table>
-  ),
-  tableHead: ({ children }: { children: React.ReactNode[] }) => (
-    <TableHeader data-test-subj="markdown-table-header">{children}</TableHeader>
-  ),
-  tableRow: ({ children }: { children: React.ReactNode[] }) => (
-    <EuiTableRow data-test-subj="markdown-table-row">{children}</EuiTableRow>
-  ),
-  tableCell: ({ children }: { children: React.ReactNode[] }) => (
-    <EuiTableRowCell data-test-subj="markdown-table-cell">{children}</EuiTableRowCell>
-  ),
-};
+export const Markdown = pure<{ raw?: string; size?: 'xs' | 's' | 'm' }>(({ raw, size = 's' }) => {
+  const markdownRenderers = {
```

Comment:
Just curious, but what was intent here with moving `markdownRenderers` into the `Markdown` component? Perf?

Reply:
The previous version of the `Markdown` component did not take `size` as a prop, which is now used by the `root` renderer. So previously, `markdownRenderers` was declared as a (non-parametrized) configuration object outside of the scope of the `Markdown` component.

Since the `root` renderer now requires access to the `size` prop, as shown in the snippet below....

```
    root: ({ children }: { children: React.ReactNode[] }) => (
      <EuiText data-test-subj="markdown-root" grow={true} size={size}>
        {children}
      </EuiText>
    ),
```

...the declaration of `markdownRenderers` moved from above the declaration of the `Markdown` component into the function body so `root` can capture the value of the `size` prop via a closure.

Reply:
Ah, okie! Thanks!